server/src/main/java/com/senlainc/warsaw/tyurin/service/CraftsmanService.java

```diff
@@ -10,20 +12,27 @@
 import com.senlainc.warsaw.tyurin.util.jsonHandlers.JsonReader;
 import com.senlainc.warsaw.tyurin.util.jsonHandlers.JsonWriter;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
+@DependencyClass
 public class CraftsmanService implements ICraftsmanService{
 
     private static CraftsmanService INSTANCE;
+    @DependencyComponent
     private ICraftsmanDAO craftsmanDAO;
+    @DependencyComponent
     private IOrderService orderService;
     private CsvReader csvReader;
     private CsvWriter csvWriter;
     private JsonReader jsonReader;
     private JsonWriter jsonWriter;
 
-    private CraftsmanService() {
+    public CraftsmanService() {
         craftsmanDAO = CraftsmanDAO.getInstance();
```

Comment:
You sure this approach will not break DI? I mean it's hard to tell what will be returned without debugger, but seems like you will get object which will be created through Constructor. 
Please double check it. and if so get rid of getInstances

Reply:
Moreover, you don;t need this instances logic, because right now you singleton instance should be in application context